runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java

```diff
@@ -112,4 +124,52 @@ public void onDecorativeObjectDespawned(DecorativeObjectDespawned event)
 		DecorativeObject object = event.getDecorativeObject();
 		streams.remove(object);
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+		if (client.getLocalPlayer().getWorldLocation().getRegionID() == TOG_REGION)
+		{
+			if (playerLowestSkill != null)
+			{
+				return;
+			}
+			if (client.getSkillExperience(Skill.HITPOINTS) > 0)
+			{
+				playerLowestSkill = getLowestPlayerSkill();
+			}
+		}
+		else
+		{
+			playerLowestSkill = null;
+		}
+	}
+
+	private Skill getLowestPlayerSkill()
+	{
+		Skill[] playerSkills = Skill.values();
+
+		Skill lowestExperienceSkill = null;
+		int lowestExperienceAmount = Integer.MAX_VALUE;
+
+		for (Skill skill : playerSkills)
+		{
+			int currentSkillExp = client.getSkillExperience(skill);
+			if (currentSkillExp < lowestExperienceAmount)
+			{
+				lowestExperienceAmount = currentSkillExp;
+				lowestExperienceSkill = skill;
+			}
+		}
+		return lowestExperienceSkill;
+	}
+
+	public int getLowestPlayerSkillLevel()

```

Comment:
Remove public

Reply:
Or even remove entire method, it is not necessary as you can just inject client to overlay and call client.getRealSkillLevel there